crates/router/src/db/merchant_connector_account.rs

```diff
@@ -374,28 +387,80 @@ impl MerchantConnectorAccountInterface for MockDb {
 
     async fn find_merchant_connector_account_by_merchant_id_and_disabled_list(
         &self,
-        _merchant_id: &str,
-        _get_disabled: bool,
+        merchant_id: &str,
+        get_disabled: bool,
     ) -> CustomResult<Vec<domain::MerchantConnectorAccount>, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let accounts = self
+            .merchant_connector_accounts
+            .lock()
+            .await
+            .iter()
+            .filter(|account: &&storage::MerchantConnectorAccount| {
+                account.merchant_id == merchant_id && account.disabled == Some(get_disabled)
+            })
+            .cloned()
+            .collect::<Vec<storage::MerchantConnectorAccount>>();
+
+        let mut output = Vec::with_capacity(accounts.len());
+        for account in accounts.into_iter() {
+            output.push(
+                account
+                    .convert(self, merchant_id, self.get_migration_timestamp())
+                    .await
+                    .change_context(errors::StorageError::DecryptionError)?,
+            )
+        }
+        Ok(output)
     }
 
     async fn update_merchant_connector_account(
         &self,
-        _this: domain::MerchantConnectorAccount,
-        _merchant_connector_account: storage::MerchantConnectorAccountUpdateInternal,
+        merchant_connector_account: domain::MerchantConnectorAccount,
+        updated_merchant_connector_account: MerchantConnectorAccountUpdateInternal,
     ) -> CustomResult<domain::MerchantConnectorAccount, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let account = self
+            .merchant_connector_accounts
+            .lock()
+            .await
+            .iter_mut()
+            .find(|account| account.id == merchant_connector_account.id.unwrap_or_default())
```

Comment:
Avoid the `unwrap_or_default()`. If the `id` field is `None`, this would end up searching for merchant connector accounts with `id` 0.